pkg/scheduler/internal/cache/cache.go

```diff
@@ -226,7 +226,7 @@ func (cache *schedulerCache) UpdateSnapshot(nodeSnapshot *Snapshot) error {
 			// Transient scheduler info is reset here.
 			node.info.TransientInfo.ResetTransientSchedulerInfo()
 		}
-		if np := node.info.Node(); np != nil {
+		if np := node.info.Node; np != nil {
```

Comment:
this is the most critical change.
If we are sure that `node.info != nil` at this point, then we can safely change usages in plugins and what not. Can you double check and give a summary of where are these objects coming from?

Reply:
When nodes/pods added or updated, we will use `n = newNodeInfoListItem(framework.NewNodeInfo())` to init cache node which assigns the new NodeInfo to `node.info`.

Another trick way to support `node.info != nil` at this point is that if so, line 221 will raise error.

Reply:
/approve
/lgtm

thanks for checking